### PR TITLE
Feature Flag final DRAFT

### PR DESCRIPTION
**Pull request by wpschaub**
Thank you for your pull request!
By completing this pull request, you agree to the [Contributing License Agreement](https://github.com/ALM-Rangers/Guidance/blob/master/.github/CLA.md).

Fixes # .

Changes proposed in this pull request:  
- Final DRAFT for the feature flag article.
- Ready to get feedabck from technical SMEs.

@ALM-Rangers/guidance

### CODE REVIEW

**Comment by wpschaub** [comment]
@mikaelkrief @edwardaf @EdwardAF-IT @OlemIur here's the final DRAFT for the feature flagging article. We need to submit it for publication by Monday to ensure we can reference it from the new article we're writing for the Connect event. Candid feedback welcome.

File: src/DevOps/phase-features-with-feature-flags.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+This topic aims to answer this and share our implementation of feature flags (FF) and A|B testing, within our production infrastructure for DevLabs extensions.
+
+## Considerations
+
+Before you introduce feature flags to your engineering process, it's important to consider the following:
+
+- Which users are you planning to target? For example, do you want to target specific users, all users, or both?
+- Would you like users to decide which features they want to use?
+- What's the value of embracing feature flags as part of your engineering process?
+- What's the cost implement feature flags in your engineering process?
```
Comment (mikefourie) [inline src/DevOps/phase-features-with-feature-flags.md]: of implementing

File: src/DevOps/phase-features-with-feature-flags.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+- How can you revert a change deployed to production without rolling back your release?
+- How can you present users with variants of a feature, to determine which one performs better?
+
+This topic aims to answer this and share our implementation of feature flags (FF) and A|B testing, within our production infrastructure for DevLabs extensions.
+
+## Considerations
+
+Before you introduce feature flags to your engineering process, it's important to consider the following:
+
+- Which users are you planning to target? For example, do you want to target specific users, all users, or both?
```
Comment (mikefourie) [inline src/DevOps/phase-features-with-feature-flags.md]: specific users or all users?

sounds odd to have 'both' targeting all and specific, are they not mutually exclusive?

File: src/DevOps/phase-features-with-feature-flags.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	- No servers - [LaunchDarkly](https://launchdarkly.com/index.html) takes care of the machines that LaunchDarkly runs on.
+	- Always on and optimized for the Internet.
+- It's integrated with Visual Studio Team Services and Team Foundation Server.
+- It's simple and cost-effective to configure and use in our community space.
+
+## Common Scenarios
+
+We have a [CI/CD pipeline ](https://blogs.msdn.microsoft.com/visualstudioalmrangers/tag/cicd-pipeline/) for every VSTS extension we're hosting on the [marketplace](https://marketplace.visualstudio.com), using a ring deployment model, and manual release approval checkpoints. The checkpoints are manual and time consuming, but necessary to minimize the chance of breaking the early-adopter and production user environments, forcing an expensive roll-back. We're looking for an engineering process in which we can continuously deploy, never have to roll-back, and with which we can fine tune the user experience.
+
+You have probably guessed it. Welcome to feature flags, which allow us to fine tune the visibility of features at **run-time** and in **production**!
```
Comment (mikefourie) [inline src/DevOps/phase-features-with-feature-flags.md]: what's the diff?

File: src/DevOps/phase-features-with-feature-flags.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+We have a [CI/CD pipeline ](https://blogs.msdn.microsoft.com/visualstudioalmrangers/tag/cicd-pipeline/) for every VSTS extension we're hosting on the [marketplace](https://marketplace.visualstudio.com), using a ring deployment model, and manual release approval checkpoints. The checkpoints are manual and time consuming, but necessary to minimize the chance of breaking the early-adopter and production user environments, forcing an expensive roll-back. We're looking for an engineering process in which we can continuously deploy, never have to roll-back, and with which we can fine tune the user experience.
+
+You have probably guessed it. Welcome to feature flags, which allow us to fine tune the visibility of features at **run-time** and in **production**! 
+
+The three core scenarios we're using are:
+
+- **Enable | disable a feature for everyone**
+
+	We would like to ship a batch of hidden features, for example new features, bug fixes, verbose logging and telemetry. Then, with a flip of a switch enable a feature for **all** users of the extension. If the feature results in an unsatisfactory user experience, we simply flip the switch.
+
```
Comment (mikefourie) [inline src/DevOps/phase-features-with-feature-flags.md]: back

File: src/DevOps/phase-features-with-feature-flags.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+- Provide run-time control of features to the end user.
+- Make changes (enable|disable features) without redeployment.
+- Introduce early testing, feedback, and experimentation. 
+- Support the quick "off" or "revert" switch for new features.
+
+In future we'll explore the option of A|B testing and allowing users to enable or disable selected features themselves. There's a lot more value to embrace than we have explored herein.
+
+## Conclusion
+
+Now that you've covered the concepts and considerations of feature flags, and our implementation of them, you should be confident to explore ways to improve your CI/CD pipelines. While feature flags come at a cost, having a game plan to manage exposed features at run-time is invaluable.
```
Comment (mikefourie) [inline src/DevOps/phase-features-with-feature-flags.md]: While feature flags come at a cost, --- you've not talked about the costs

File: src/DevOps/phase-features-with-feature-flags.md (hunk 1 above)
Comment (wpschaub) [reply src/DevOps/phase-features-with-feature-flags.md]: Fixed.

File: src/DevOps/phase-features-with-feature-flags.md (hunk 3 above)
Comment (wpschaub) [reply src/DevOps/phase-features-with-feature-flags.md]: There is none. Removed "run-time".

File: src/DevOps/phase-features-with-feature-flags.md (hunk 4 above)
Comment (wpschaub) [reply src/DevOps/phase-features-with-feature-flags.md]: Done

**Pull request merged**